src/backend/commands/createas.c

```diff
@@ -1275,3 +1267,91 @@ is_equijoin_condition(OpExpr *op)
 
 	return false;
 }
+
+/*
+ * check_aggregate_supports_ivm
+ *
+ * Check if the given aggregate function is supporting
+ */
+static void
+check_aggregate_supports_ivm(Oid aggfnoid)
+{
+	switch (aggfnoid)
+	{
+		/* count */
+		case F_AGG_COUNT_ANY:
+		case F_AGG_COUNT_:
+
+		/* sum */
+		case F_AGG_SUM_INT8:
+		case F_AGG_SUM_INT4:
+		case F_AGG_SUM_INT2:
+		case F_AGG_SUM_FLOAT4:
+		case F_AGG_SUM_FLOAT8:
+		case F_AGG_SUM_MONEY:
+		case F_AGG_SUM_INTERVAL:
+		case F_AGG_SUM_NUMERIC:
+
+		/* avg */
+		case F_AGG_AVG_INT8:
+		case F_AGG_AVG_INT4:
+		case F_AGG_AVG_INT2:
+		case F_AGG_AVG_NUMERIC:
+		case F_AGG_AVG_FLOAT4:
+		case F_AGG_AVG_FLOAT8:
+		case F_AGG_AVG_INTERVAL:
+
+		/* min */
+		case F_AGG_MIN_ANYARRAY:
+		case F_AGG_MIN_INT8:
+		case F_AGG_MIN_INT4:
+		case F_AGG_MIN_INT2:
+		case F_AGG_MIN_OID:
+		case F_AGG_MIN_FLOAT4:
+		case F_AGG_MIN_FLOAT8:
+		case F_AGG_MIN_DATE:
+		case F_AGG_MIN_TIME:
+		case F_AGG_MIN_TIMETZ:
+		case F_AGG_MIN_MONEY:
+		case F_AGG_MIN_TIMESTAMP:
+		case F_AGG_MIN_TIMESTAMPTZ:
+		case F_AGG_MIN_INTERVAL:
+		case F_AGG_MIN_TEXT:
+		case F_AGG_MIN_NUMERIC:
+		case F_AGG_MIN_BPCHAR:
+		case F_AGG_MIN_TID:
+		case F_AGG_MIN_ANYENUM:
+		case F_AGG_MIN_INET:
+		case F_AGG_MIN_PG_LSN:
+
+		/* max */
+		case F_AGG_MAX_ANYARRAY:
+		case F_AGG_MAX_INT8:
+		case F_AGG_MAX_INT4:
+		case F_AGG_MAX_INT2:
+		case F_AGG_MAX_OID:
+		case F_AGG_MAX_FLOAT4:
+		case F_AGG_MAX_FLOAT8:
+		case F_AGG_MAX_DATE:
+		case F_AGG_MAX_TIME:
+		case F_AGG_MAX_TIMETZ:
+		case F_AGG_MAX_MONEY:
+		case F_AGG_MAX_TIMESTAMP:
+		case F_AGG_MAX_TIMESTAMPTZ:
+		case F_AGG_MAX_INTERVAL:
+		case F_AGG_MAX_TEXT:
+		case F_AGG_MAX_NUMERIC:
+		case F_AGG_MAX_BPCHAR:
+		case F_AGG_MAX_TID:
+		case F_AGG_MAX_ANYENUM:
+		case F_AGG_MAX_INET:
+		case F_AGG_MAX_PG_LSN:
+			return;
+
+		default:
+		{
+			const char *aggname = get_func_name(aggfnoid);
+			elog(ERROR, "aggregate function %s is not supported", aggname);
```

Comment:
Shouldn't check_aggregate_supports_ivm() return ok/ng to caller so that the caller can handle the error?

Reply:
This makes sense. I fixed this as you suggested.